tests/test_script_helper.py

```diff
@@ -1,222 +1,284 @@
 
-import neuron_reduce
-subtree_reductor =  neuron_reduce.subtree_reductor
-
-APICAL = "apical"
-SOMATIC = "somatic"
+from neuron_reduce import subtree_reductor
 
 #############################
-# simulation parameters - to be adjusted by the user:
-TSTOP = 1500  # ms
-STEPS_PER_MS = 10
-DT = 0.1
+# simulation parameters - to be adjusted by the user
+SIM_PARAMS = {'tstop': 1500,  # ms
+              'steps_per_ms': 10,
+              'dt': 0.1,
+              }
 
 SPIKE_BEGIN_TIME = 10  # ms
 WEIGHT = 0.0008  # gbar = 0.8 nS
 
-ICLAMP_DURATION = 1500 # ms
-ICLAMP_DELAY = 10
-ICLAMP_AMPLITUDE = 0 #nA
+ICLAMP_PARAMS = {'duration': 1500,  # ms
+                 'delay': 10,
+                 'amplitude': 0,  # nA
+                 }
+
+EXCITATORY_STIMULI_INTERVAL = 1000  # ms   -  100 ms interval is 10 hz
+EXCITATORY_STIMULI_NUMBER = 15
+FREQUENCY_OF_INHIBITORY_SYNAPSES_IS_HIGHER_BY = 10
+
+INHIBITORY_STIMULI_INTERVAL = (float(EXCITATORY_STIMULI_INTERVAL) /
+                               FREQUENCY_OF_INHIBITORY_SYNAPSES_IS_HIGHER_BY)  # ms
+INHIBITORY_STIMULI_NUMBER = int(float(ICLAMP_PARAMS['duration']) / INHIBITORY_STIMULI_INTERVAL)
+
+SYNAPSE_PARAMS = {'excitatory': {'e_syn': 0,  # mV
+                                 'tau_1': 0.3,  # ms (like AMPA)
+                                 'tau_2': 1.8,  # ms (like AMPA)
+                                 'stimuli_interval': EXCITATORY_STIMULI_INTERVAL,
+                                 'stimuli_number': EXCITATORY_STIMULI_NUMBER,
+                                 },
+                  'inhibitory': {'e_syn': -86,
+                                 'tau_1': 1.,  # ms
+                                 'tau_2': 8.,  # ms
+                                 'stimuli_interval': INHIBITORY_STIMULI_INTERVAL,
+                                 'stimuli_number': INHIBITORY_STIMULI_NUMBER,
+                                 },
+                  }
+
+PERCENTAGE_OF_EXCITATORY_SYNAPSES = 85.
+STIMULI_NOISE = 1
+SEED_FOR_RANDOM_SYNAPSE_CLASSIFICATION = 1
 
-EXCITATORY_E_SYN = 0 #mV
-EXCITATORY_TAU_1 = 0.3 #ms  (like AMPA)
-EXCITATORY_TAU_2 = 1.8 #ms  (like AMPA)
 
-INHIBITORY_E_SYN = -86
-INHIBITORY_TAU_1 = 1 #ms
-INHIBITORY_TAU_2 = 8 #ms
+def abs_diff(a, b):
+    '''return summed absolute difference'''
+    return np.sum(np.abs(a - b))
 
-PERCENTAGE_OF_EXCITATORY_SYNAPSES = 85               
-FREQUENCY_OF_INHIBITORY_SYNAPSES_IS_HIGHER_BY = 10
 
-EXCITATORY_STIMULI_INTERVAL = 1000  #ms   -  100 ms interval is 10 hz
-EXCITATORY_STIMULI_NUMBER = 15           
+def rmsd(reduced, control):
+    '''find root mean squared deviation'''
+    diff = reduced - control
+    return np.sqrt(np.dot(diff, diff) / reduced.size)
 
-INHIBITORY_STIMULI_INTERVAL = float(EXCITATORY_STIMULI_INTERVAL) / FREQUENCY_OF_INHIBITORY_SYNAPSES_IS_HIGHER_BY  # ms  
-INHIBITORY_STIMULI_NUMBER = int(float(ICLAMP_DURATION) / INHIBITORY_STIMULI_INTERVAL)   
 
-STIMULI_NOISE = 1
-#############################
+@contextmanager
+def cd(path):
```

Comment:
Isn't cd part of ipython functions? Maybe it would be better to change the function name to cd_yield?

Reply:
true, not sure you would have a collision here, but I'll change it